### ISSUE TITLE
The implement of rank-aware contrastive loss

### ISSUE THREAD

**Issue by KonataMyLove** [wjun0830/QD-DETR#21]
Thank you for your open-source code.
Here I have a question about the code of rank-aware contrastive loss metioned in the Section3.4 of the paper:

  This is the rank-aware contrastive loss formula listed in the paper,
![图片](https://github.com/wjun0830/QD-DETR/assets/90315108/a706e9e8-3bb7-4c6f-a68e-dd67de55955d)
  

  This is the code of rank-aware contrastive loss computation in the qd_detr/model.py:
            # softmax
            exp_logits = torch.exp(logits)
            **log_prob = logits - torch.log(exp_logits.sum(1, keepdim=True) + 1e-6)**
            mean_log_prob_pos = (pos_mask * log_prob * vid_token_mask).sum(1) / (pos_mask.sum(1) + 1e-6)
            loss = - mean_log_prob_pos * batch_drop_mask
            loss_rank_contrastive = loss_rank_contrastive + loss.mean()

  I don't think the code implement matched the formula? Especially the bolded code makes me confused.
If I get it wrong, please inform me. Thank you for your time.

**Comment by wjun0830** [comment]
For the contrastive equation and the implementation, you can find descriptions in supervised contrastive Learning paper and [repository] (https://github.com/HobbitLong/SupContrast/issues/116).
For the rank, there is a for loop to implement it.

**Comment by KonataMyLove** [comment]
Thanks. I rewrote the formula according to the repository and the code, it does match the paper. Problem solved.
